pkg/scheduler/objects/queue.go

```diff
@@ -369,17 +372,74 @@ func (sq *Queue) applyConf(conf configs.QueueConfig, silence bool) error {
 
 	// Load the max & guaranteed resources and maxApps for all but the root queue
 	if sq.Name != configs.RootQueue {
+		oldMaxResource := sq.maxResource
 		if err = sq.setResourcesFromConf(conf.Resources); err != nil {
 			return err
 		}
 		sq.maxRunningApps = conf.MaxApplications
 		sq.updateMaxRunningAppsMetrics()
+		sq.setPreemptionSettings(oldMaxResource, conf)
 	}
 
 	sq.properties = conf.Properties
 	return nil
 }
 
+// setPreemptionSettings Set Quota change preemption settings
+func (sq *Queue) setPreemptionSettings(oldMaxResource *resources.Resource, conf configs.QueueConfig) {
+	newMaxResource, err := resources.NewResourceFromConf(conf.Resources.Max)
+	if err != nil {
+		log.Log(log.SchedQueue).Error("parsing failed on max resources this should not happen",
+			zap.String("queue", sq.QueuePath),
+			zap.Error(err))
+		return
+	}
+	reset := false
+	set := false
+	switch {
+	// Set max res earlier but not now
+	case resources.IsZero(newMaxResource) && !resources.IsZero(oldMaxResource):
+		reset = true
+		// Set max res now but not earlier
+	case !resources.IsZero(newMaxResource) && resources.IsZero(oldMaxResource) && conf.Preemption.Delay != 0:
+		set = true
+		// Set max res earlier and now as well
+	default:
+		switch {
+		// Quota decrease
+		case resources.StrictlyGreaterThan(oldMaxResource, newMaxResource) && conf.Preemption.Delay != 0:
+			set = true
+			// Quota remains as is but delay has changed
+		case resources.Equals(oldMaxResource, newMaxResource) && conf.Preemption.Delay != 0 && sq.quotaChangePreemptionDelay != conf.Preemption.Delay:
+			sq.quotaChangePreemptionDelay = conf.Preemption.Delay
+			sq.quotaChangePreemptionTimer.Reset(time.Duration(sq.quotaChangePreemptionDelay))
```

Comment:
That's why I don't like having a Timer object here.  We need to rely on proper understanding of how Timer works in Go, see: https://antonz.org/timer-reset/.